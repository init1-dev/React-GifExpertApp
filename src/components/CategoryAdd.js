import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = ({ categories, setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    };

    const handleSubmit = ( e ) => {
        e.preventDefault();

        // This could be done as helper function
        if( inputValue.trim().length > 2 && !undefined ) {
            if(categories.includes(inputValue)) {
                alert('This category already exist');
                setInputValue('');
            } else {
                setCategories( cats => [ inputValue, ...cats ] );
                setInputValue('');
            }
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                name="categories"
                id="categories"
                placeholder="Search Categories"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

export default CategoryAdd;

CategoryAdd.propTypes = {
    categories: PropTypes.array.isRequired,
    setCategories: PropTypes.func.isRequired
}