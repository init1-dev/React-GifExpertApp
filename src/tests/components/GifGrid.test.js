import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid.js />', () => {

    const category = 'One piece';
    
    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/imagen.jpg',
            title: 'Una imagen'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( 1 );
    })
    
});