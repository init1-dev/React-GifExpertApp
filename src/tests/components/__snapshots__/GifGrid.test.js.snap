// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid.js /> Debe mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One piece
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Una imagen"
      url="https://localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid.js /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One piece
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
